tamr_unify_client/client.py

```diff
@@ -140,3 +140,15 @@ def datasets(self):
         :rtype: :class:`~tamr_unify_client.models.DatasetCollection`
         """
         return self._datasets
+
+    def __repr__(self):
+        # intentionally leaving out auth in order to avoid security concerns
+        return (
+            f"{self.__class__.__module__}."
+            f"{self.__class__.__qualname__}("
+            f"host={self.host!r}, "
+            f"protocol={self.protocol!r}, "
+            f"port={self.port!r}, "
+            f"base_path={self.base_path!r}"
+            f")@0x{id(self):x}"
```

Comment:
@DerrickRice Genuinely not sure how I feel about including the object ID in the `repr`. I can see how it could be needlessly confusing to beginners, but could also see how it could help distinguish different instances.

Do you know if showing the object ID in the `repr` is a common pattern?

Reply:
It is not a common pattern and I am not attached to it. It does trade brevity/simplicity for specificity.

I suspect that it's more likely that the overall cost of including it (i.e. confusion and verbosity) is higher than the risk of confusion when two objects have the same repr but aren't the same object.

It is common for `__repr__` to not distinguish between two exactly similar objects.

Let's remove it?

Reply:
👍 Yes, I agree. Let's remove it.